### PR TITLE
create Add to cart and buy case

### PR DESCRIPTION
**Pull request by vdemnyk**

### CODE REVIEW

File: base/webElement.js [hunk 1]
```diff
@@ -21,6 +21,16 @@ class WebElement {
         let ec = protractor.ExpectedConditions;
         await browser.wait(ec.visibilityOf(this.protractorElement), time); 
     }
+
+    async mouseMove() {
```
Comment (nbabii) [inline base/webElement.js]: let's call it `mouseHover`

File: elemnets/textInput.js [hunk 2]
```diff
@@ -10,6 +11,15 @@ class WebTextInput extends WebElement{
         console.log('Clearing field');
         await this.protractorElement.clear();
     }
+
+    async check() {
+        console.log('Checking checkbox');
+        let checked = await this.protractorElement.isSelected();
```
Comment (nbabii) [inline elemnets/textInput.js]: Please test this method if it works correctly in the case with already selected checkbox

File: elemnets/textInput.js [hunk 3]
```diff
@@ -10,6 +11,15 @@ class WebTextInput extends WebElement{
...
+        if(!checked){
+            await this.protractorElement.click();
+        }
+    }
+
```
Comment (nbabii) [inline elemnets/textInput.js]: Create a second method `uncheck`, or handle unchecking possibility in previous method

File: pages/category.page.js [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+    }
+
+    async clickAddToCart() {
+        await allure.createStep('Click on "Add to cart" button', async () => {
+            await this.getAddToCartElement().click();
+        })();
+    }
+
+    async cartModalWindowIsOpened() {
+        await browser.wait(protractor.ExpectedConditions.visibilityOf(this.getCartModalWindowElement().protractorElement), 20000);
```
Comment (nbabii) [inline pages/category.page.js]: too complicated, you can just use `this.getCartModalWindowElement().waitForVisible (20000)`

**Review by nbabii — CHANGES_REQUESTED** [review]

File: elemnets/textInput.js (hunk 2 above)
Comment (nbabii) [reply elemnets/textInput.js]: move to checkbox file

File: base/webElement.js (hunk 1 above)
Comment (vdemnyk) [reply base/webElement.js]: Done

File: elemnets/textInput.js (hunk 2 above)
Comment (vdemnyk) [reply elemnets/textInput.js]: Done

File: elemnets/textInput.js (hunk 3 above)
Comment (vdemnyk) [reply elemnets/textInput.js]: Done

File: pages/category.page.js (hunk 4 above)
Comment (vdemnyk) [reply pages/category.page.js]: Done

File: base/webPage.js [hunk 5]
```diff
@@ -7,7 +7,7 @@ class WebPage {
     }
 
     async waitForPageLoaded() {
-        await browser.wait(protractor.ExpectedConditions.visibilityOf(this.getBaseElement().protractorElement), 20000);
+        await browser.wait(protractor.ExpectedConditions.visibilityOf(this.getBaseElement().protractorElement), 30000);
```
Comment (nbabii) [inline base/webPage.js]: use `this.getBaseElement().waitForVisible()`

File: pages/cart.page.js [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+    async typeQuantity(number) {
+        await allure.createStep(`Clear "Qty" field, type ${number} in "Qty" field`, async () => {
+            await this.getQuantityFieldElement().clear();
+            await this.getQuantityFieldElement().sendKeys(number);
+        })();   
+    }
+
+    async checkQuantity() {
+        await this.getQuantityValueElement().waitForPresent();
+        return await this.getQuantityValueElement().isPresent();
```
Comment (nbabii) [inline pages/cart.page.js:27]: not logical part of code, it will never return `false` as it will fall on previous line

**Review by nbabii — CHANGES_REQUESTED** [review]

File: base/webPage.js (hunk 5 above)
Comment (vdemnyk) [reply base/webPage.js]: Done

File: pages/cart.page.js (hunk 6 above)
Comment (vdemnyk) [reply pages/cart.page.js:27]: Done

**Pull request merged**